packages/mdc-menu/foundation.ts

```diff
@@ -132,41 +130,36 @@ export class MDCMenuFoundation extends MDCFoundation<MDCMenuAdapter> {
   }
 
   /**
-   * Handles toggling the selected classes in a selection group when a selection is made.
+   * Selects the list item at `index` within the menu.
+   * @param index Index of list item within the menu.
    */
-  private handleSelectionGroup_(selectionGroup: Element, index: number) {
-    // De-select the previous selection in this group.
-    const selectedIndex = this.adapter_.getSelectedElementIndex(selectionGroup);
-    if (selectedIndex >= 0) {
-      this.adapter_.removeAttributeFromElementAtIndex(selectedIndex, strings.ARIA_SELECTED_ATTR);
-      this.adapter_.removeClassFromElementAtIndex(selectedIndex, cssClasses.MENU_SELECTED_LIST_ITEM);
+  setSelectedIndex(index: number) {
+    if (!this.isIndexInRange_(index)) {
```

Comment:
Since this function is validating the index and throws error when index is invalid, let's call this `validateIndex_`. The return statement will never be executed with current structure.

```
this.isIndexValid_(index);
```

Reply:
sorry i'm confused. Do you want to call it `isIndexValid_` or `validateIndex_`?

Reply:
I'm suggesting, `this.validateIndex_()`.


As per current implementation `isIndexValid_()` does not always return boolean.

Reply:
Ah sorry I was confused...current implementation is `isIndexInRange_`. I'll change to validateIndex and return boolean.

Reply:
Sorry for not being clearer. `isIndexInRange_` seems perfectly fine if the function is just returning a boolean without any side-effects (i.e., throwing an error).

My earlier comment was that this if condition will never fail, because the method throws JS error instead of returning `false` value. (I noticed this in [coverage report](https://codecov.io/gh/material-components/material-components-web/src/d2ea93dcfa07e43620615f960242687fc66514d6/packages/mdc-menu/foundation.ts#L94)).

Reply:
OH - I understood what you meant after I implemented this. I think I changed it correctly. I just need to update tests.